ctru-rs/src/romfs.rs

```diff
@@ -11,39 +11,39 @@
 //! ```
 
 use std::ffi::CStr;
-use std::sync::atomic::{AtomicBool, Ordering};
+use std::lazy::SyncLazy;
+use std::sync::Mutex;
 
-use crate::error::Error;
+use crate::services::ServiceHandler;
 
 #[non_exhaustive]
-pub struct RomFS;
+pub struct RomFS {
+    _service_handler: ServiceHandler,
+}
 
-static ROMFS_ACTIVE: AtomicBool = AtomicBool::new(false);
+static ROMFS_ACTIVE: SyncLazy<Mutex<usize>> = SyncLazy::new(|| Mutex::new(0));
 
 impl RomFS {
     pub fn init() -> crate::Result<Self> {
-        match ROMFS_ACTIVE.compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst) {
-            Ok(_) => {
+        let _service_handler = ServiceHandler::new(
+            &ROMFS_ACTIVE,
+            true,
```

Comment:
We should probably set this to false.

Reply:
Why? The start code isn’t run again.

Reply:
That's fair. I guess if we add `&mut self` functions we should switch it to false, but it's fine for now.